packages/medusa/src/loaders/plugins.ts

```diff
@@ -371,22 +374,12 @@ export async function registerServices(
         throw new Error(message)
       }
 
-      if (isPaymentService(loaded.prototype)) {
-        // Register our payment providers to paymentProviders
-        container.registerAdd(
-          "paymentProviders",
-          asFunction((cradle) => new loaded(cradle, pluginDetails.options))
-        )
+      const context = { container, pluginDetails, registrationName: name }
 
-        // Add the service directly to the container in order to make simple
-        // resolution if we already know which payment provider we need to use
-        container.register({
-          [name]: asFunction(
-            (cradle) => new loaded(cradle, pluginDetails.options)
-          ),
-          [`pp_${loaded.identifier}`]: aliasTo(name),
-        })
-      } else if (loaded.prototype instanceof OauthService) {
+      registerPaymentServiceFromClass(loaded, context)
+      registerPaymentProcessorFromClass(loaded, context)
```

Comment:
comment: These are quite nice. Should we include utils for all types of classes?

Reply:
yes definitely, just did not want to include all of them in that pr